### PR TITLE
add a simple sizeof checker to compare nim & c types

### PR DESCRIPTION
**Pull request by arnetheduck**

### CODE REVIEW

File: compiler/ccgtypes.nim [hunk 1]
```diff
@@ -701,7 +701,10 @@ proc getTypeDescAux(m: BModule, typ: PType, check: var IntSet): Rope =
       idTablePut(m.typeCache, t, result) # always call for sideeffects:
       let recdesc = if t.kind != tyTuple: getRecordDesc(m, t, result, check)
                     else: getTupleDesc(m, t, result, check)
-      if not isImportedType(t): add(m.s[cfsTypes], recdesc)
+      if not isImportedType(t):
+        add(m.s[cfsTypes], recdesc)
+      elif tfIncompleteStruct notin t.flags:
```
Comment (Araq) [inline compiler/ccgtypes.nim]: and isDefined("nimcheckabi")

File: lib/nimbase.h [hunk 2]
```diff
@@ -459,3 +459,11 @@ typedef int Nim_and_C_compiler_disagree_on_target_architecture[sizeof(NI) == siz
 #elif defined(__FreeBSD__)
 #  include <sys/types.h>
 #endif
+
+/* Compile with -t:-DNIM_CHECK_ABI to enable */
```
Comment (Araq) [inline lib/nimbase.h]: That's not how we do things. Make the #define unconditional and use -d:nimCheckAbi
Also this features needs to be documented in doc/nimc.rst

**Comment by arnetheduck** [comment]
fixed.

File: compiler/ccgtypes.nim [hunk 3]
```diff
@@ -267,6 +271,11 @@ proc getSimpleTypeDesc(m: BModule, typ: PType): Rope =
     result = getSimpleTypeDesc(m, lastSon typ)
   else: result = nil
 
+  if result != nil and typ.isImportedType():
```
Comment (Araq) [inline compiler/ccgtypes.nim]: What's this section doing here? Don't muck with the typeCache please.

**Pull request merged**